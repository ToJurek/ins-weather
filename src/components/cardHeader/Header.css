.header {
    font-size: 2em;
}

.map-marker {
    margin-right: 10px;
}
