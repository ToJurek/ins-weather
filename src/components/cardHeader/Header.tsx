import React from 'react';
import './Header.css';
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faMapMarkerAlt} from "@fortawesome/free-solid-svg-icons";

interface Props {
    city: string
}

function Header(props: Props) {
    return (
        <div className={'header'}>
            <FontAwesomeIcon icon={faMapMarkerAlt} className={'map-marker'}/>
            <span className={'city'}>{props.city}</span>
        </div>
    )
}

export default Header;
