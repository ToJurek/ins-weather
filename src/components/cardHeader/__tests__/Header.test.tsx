import React from "react";
import Enzyme, {mount} from 'enzyme';
import Header from "../Header";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";


Enzyme.configure({adapter: new Adapter()});

describe('header', function () {
    let wrapper: any
    beforeEach(() => {
        wrapper = mount(<Header city={'city-name'}/>)
    })
    test('should match to snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    test('should render city name correctly', () => {
        expect(wrapper.find('.city').text()).toBe('city-name')
    })

});
