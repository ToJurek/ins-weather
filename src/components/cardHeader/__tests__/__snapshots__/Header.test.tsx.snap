// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should match to snapshot 1`] = `
<Header
  city="city-name"
>
  <div
    className="header"
  >
    <FontAwesomeIcon
      border={false}
      className="map-marker"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            384,
            512,
            Array [],
            "f3c5",
            "M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z",
          ],
          "iconName": "map-marker-alt",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-map-marker-alt fa-w-12 map-marker"
        data-icon="map-marker-alt"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </FontAwesomeIcon>
    <span
      className="city"
    >
      city-name
    </span>
  </div>
</Header>
`;
